src/librawspeed/io/Buffer.h

```diff
@@ -22,15 +22,19 @@
 #pragma once
 
 #include "common/Common.h" // for uchar8, uint32, uint64
+#include "io/IOException.h"// for ThrowIOE
 #include "io/Endianness.h" // for getByteSwapped
 #include <algorithm>       // for swap
 
 namespace RawSpeed {
 
-// All file maps have this much space extra at the end. This is useful for
-// BitPump* that needs to have a bit of extra space at the end to be able to
-// do more efficient reads without crashing
-#define FILEMAP_MARGIN 16UL
+// This allows to specify the nuber of bytes that each Buffer needs to
+// allocate additionally to be able to remove one runtime bounds check
+// in BitSream::fill. There are two options:
+// 0 : allocate exactly as much data as required
+// 4 : add minimum number of bytes to keep all BitStream implementations happy
+//     this disables the bounds checking, saves about 1% on modern CPUs
+#define BUFFER_PADDING 0
```

Comment:
> The code still does assume that the buffer has PADDING padding size at the end, regardless of how it said buffer was acquired, yes?

I'm not sure I understand you question. I'll try answering it anyway: The code does assume the buffer has 0 bytes padding at the end (i.e. _no_ padding), regardless of how that buffer was acquired.